packages/flutter_tools/lib/src/observatory.dart

```diff
@@ -229,9 +216,10 @@ class Observatory {
   }
 
   Future<Response> flutterExit(String isolateId) {
-    return peer.sendRequest('ext.flutter.exit', <String, dynamic>{
-      'isolateId': isolateId
-    }).then((dynamic result) => new Response(result));
+    return peer
+      .sendRequest('ext.flutter.exit', <String, dynamic>{ 'isolateId': isolateId })
+      .then((dynamic result) => new Response(result))
+      .timeout(new Duration(seconds: 2), onTimeout: () => null);
```

Comment:
I think this is ok here, it that what I want is for a non-response from the other side of the wire to not block calls to this API indefinitely. Having the flutter.exit call actually succeed after the timeout is ok.